### PR TITLE
Update win-ci-pipeline.yml: enable xnnpack tests

### PR DESCRIPTION
**Pull request by snnn**
1. Enable xnnpack test
2. Change TSA database name from onnxruntime_master to onnxruntime_main. This is a leftover of renaming the "master" branch to "main"
3. Add two static analysis jobs for WinML and DML
4. Rename the machine pool "aiinfra-dml-winbuild" to "onnxruntime-Win2019-GPU-dml-A10", so that the internal and public ADO instances use the same machine pool name.
5. Move Windows GPU CI build pipeline from "onnxruntime-Win2022-GPU-T4" to "onnxruntime-Win2022-GPU-A10" machine pool, because we do not have enough T4 GPUs.

### CODE REVIEW

File: onnxruntime/test/providers/xnnpack/xnnpack_basic_test.cc [hunk 1]
```diff
@@ -255,7 +255,7 @@ TEST(XnnpackEP, DISABLED_TestQDQConvS8S8) {  //  [ONNXRuntimeError] : 9 : NOT_IM
   RunModelTestWithPath(ort_model_path, "xnnpack_qdq_test_graph_conv_s8s8", graph_verify, 0.2f);
 }
 
-TEST(XnnpackEP, TestQDQConvS8S8_per_channel) {
+TEST(XnnpackEP, DISABLED_TestQDQConvS8S8_per_channel) {
```
[the remark this answers is not in the document]
Comment (snnn) [reply onnxruntime/test/providers/xnnpack/xnnpack_basic_test.cc:258]: Then I will wait you.

[the remark this answers is not in the document]
Comment (wejoncy) [reply onnxruntime/test/providers/xnnpack/xnnpack_basic_test.cc:258]: https://github.com/microsoft/onnxruntime/pull/16281 is merged.

**Review by mszhanyi — APPROVED** [review]

**Pull request merged**